src/marshmallow/fields.py

```diff
@@ -730,7 +733,38 @@ def _deserialize(self, value, attr, data, **kwargs) -> typing.List[typing.Any]:
                 errors.update({idx: error.messages})
         if errors:
             raise ValidationError(errors, valid_data=result)
-        return result
+        if self.deserialization_type is list:
+            return result
+        return self.deserialization_type(result)  # type: ignore
+
+
+class List(Iterable):
+    """A list field, composed with another `Field` class or
+    instance.
+
+    Example: ::
+
+        numbers = fields.List(fields.Float())
+
+    :param cls_or_instance: A field class or instance.
+    :param kwargs: The same keyword arguments that :class:`Field` receives.
+
+    .. versionchanged:: 2.0.0
+        The ``allow_none`` parameter now applies to deserialization and
+        has the same semantics as the other fields.
+
+    .. versionchanged:: 3.0.0rc9
+        Does not serialize scalar values to single-item lists.
+
+    .. versionchanged:: 3.12.0
+        Inherits from Iterable instead of Field.
```

Comment:
I don't think this is worth mentioning.

`Mapping` docstring mentions the version in which it was added, but `Dict` does not mention when it inherited from it.